api/v1beta2/foundationdbcluster_types.go

```diff
@@ -2528,3 +2533,8 @@ func (cluster *FoundationDBCluster) Validate() error {
 func (cluster *FoundationDBCluster) IsTaintFeatureDisabled() bool {
 	return len(cluster.Spec.AutomationOptions.Replacements.TaintReplacementOptions) == 0
 }
+
+// Get MaxZonesWithUnavailablePods returns the maximum number of zones that can have unavailable pods.
+func (cluster *FoundationDBCluster) GetMaxZonesWithUnavailablePods() int {
+	return pointer.IntDeref(cluster.Spec.MaxZonesWithUnavailablePods, 0)
```

Comment:
```suggestion
	return pointer.IntDeref(cluster.Spec.MaxZonesWithUnavailablePods, math.MaxInt)
```

If we just set this value to the maximum integer value we don't have to make any special cases around `0`.